x-pack/plugins/alerting/server/rules_client/lib/validate_rule_aggregation_fields.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import type { AggregationsAggregationContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+const INVALID_RULE_AGG_TERMS = [
+  'kibana.alert.rule.consumer',
+  'kibana.alert.rule.producer',
+  'alert.attributes.consumer',
```

Comment:
we should add these ones:
 `alert.attributes.producer`
 `alert.attributes.apiKeyOwner` 
`alert.attributes.meta.versionApiKeyLastmodified`
we should also add these too
```ts
const fieldsToExcludeFromPublicApi: Array<keyof SanitizedRule> = [
  'monitoring',
  'mapped_params',
  'snoozeSchedule',
  // activeSnoozes does not need to be part of the list because it gets build on the fly
  'activeSnoozes',
];
```

Reply:
We talked about that and we think th elist should look like that
```ts
const INVALID_RULE_AGG_TERMS = [
  'alert.attributes.producer',
  'alert.attributes.consumer',
 'alert.attributes.apiKey',
  'alert.attributes.apiKeyOwner',
  'alert.attributes.meta.versionApiKeyLastmodified'
```